src/react.tsx

```diff
@@ -15,23 +38,36 @@ export function connect<StoreState extends object>(store: StoreDefinition<StoreS
   ): React.ComponentClass<Diff<PropsWithStore, { store: Store<StoreState> }>> {
 
     type State = {
-      store: StoreSnapshot<StoreState>
-      subscription: Subscription
+      store: StoreSnapshotWrapper<StoreState>
     }
 
     return class extends React.Component<Diff<PropsWithStore, { store: Store<StoreState> }>, State> {
       static displayName = `withStore(${getDisplayName(Component)})`
+      const subscriptions: Map<keyof StoreState, Subscription> = new Map()
+
+      const _onGet = (field: keyof StoreState) => {
+        if (this.subscriptions.has(field)) {
+          return
+        }
+        this.subscriptions.set(
+          field,
+          store.on(field).subscribe(
+            (value) => {
+              if (equals(value, this.state.store.get(field))) {
+                return
+              }
+              this.setState({
+                store: new StoreSnapshotWrapper(store.getCurrentSnapshot(), this._onGet)
```

Comment:
Do you mean `_onGet`?

Let's be sure to add some tests as part of this PR:

1. A component that depends on `store.get('a')` should re-render when `a` updates but not when `b` updates
2. A component that depends on `store.get('a')` and after the first render also depends on `store.get('b')`, should re-render when `a` updates at first, and should start updating when `b` updates too. Eg.
  ```jsx
  render() {
    <button onClick={store.set('a')(store.get('a') + 1)} />
    {store.get('a') > 0
      ? store.get('b') // We should start listening on this after the user clicks the button
      : <span />
    }
  ```
3. (2), but even after we no longer depend on `b`, we should keep re-rendering (this is a case we'll want to optimize at some point). Eg.
  ```jsx
  render() {
    <button onClick={store.set('a')(store.get('a') + 1)} />
    <button onClick={store.set('a')(store.get('a') - 1)} />
    {store.get('a') > 0
      ? store.get('b') // We should keep listening on this after the user clicks the first button, then the second button
      : <span />
    }
  ```
4. Repeat 1-3 for `.getState`, and for `.get` and `.getState` calls outside of `render` (eg. in a different method on the React component).

Reply:
My mistake - I didn't realize that `_onGet` is actually an instance method, not a local variable. See my comment above.